src/core/Context.cc

```diff
@@ -87,24 +87,24 @@ const Value& Context::lookup_variable(const std::string& name, const Location& l
   return *result;
 }
 
-boost::optional<CallableFunction> Context::lookup_function(const std::string& name, const Location& loc) const
+boost::optional<CallableFunction> Context::lookup_function(const Identifier& name, const Location& loc) const
 {
-  if (is_config_variable(name)) {
+  if (name.is_config_variable()) {
     return session()->lookup_special_function(name, loc);
   }
   for (const Context *context = this; context != nullptr; context = context->getParent().get()) {
     boost::optional<CallableFunction> result = context->lookup_local_function(name, loc);
     if (result) {
-      return result;
+      return std::move(result);
```

Comment:
I wouldn't worry about this, but just a note to whomever is reading this that at some point we should probably go through our usage of std::move(). In this case, moving shouldn't be needed due to NRVO copy elision, although the effect is likely negligible.